integration-tests/environments/react-native/metro.config.js

```diff
@@ -20,36 +20,81 @@ const path = require("path");
 const fs = require("fs");
 const exclusionList = require("metro-config/src/defaults/exclusionList");
 
-// Read the localDependencies property of package.json and resolve the relative path values
-const appPackageJson = readJson(__dirname, "package.json");
-const localDependencies = Object.fromEntries(
-  Object.entries(appPackageJson.localDependencies || {}).map(([localName, relativePath]) => [
-    localName,
-    path.resolve(__dirname, relativePath),
-  ]),
-);
-
 function readJson(...pathSegemnts) {
   const filePath = path.join(...pathSegemnts);
   return JSON.parse(fs.readFileSync(filePath, "utf8"));
 }
 
-const watchFolders = Object.values(localDependencies);
+function getLinkedDependencies(packagePath, recursive = true, exclude = new Set()) {
```

Comment:
Could we add a high level description of what this process is doing/why it exists in a comment, as it's not necessarily obvious? Just to make it clear for anyone new to the codebase, without having to read the details. Something like

```
// Metro bundler does not support symlinks. In order to link to our local version 
// of Realm JS in a way which Metro can follow, we need to instead add the module 
// to `resolver.blockList`, and then in `resolver.extraNodeModules`, override the 
// locations for any symlinked modules by specifying a mapping from their name 
// to their actual location on disk.

// In order to achieve this, we recursively check if each file/directory in `node_modules` 
// is a symlink or not. If it is, and if it is specified in our main `package.json` (i.e. 
// it is a direct dependency of our app), we add it to the `blockList` and 
// add a mapping for it. We also add it to the list of watched folders, so
// that any changes to it are picked up by hot reloading.
```

Reply:
> Do you think it would make sense to move this logic to somewhere shared.

Definitely. If other apps would be copy-pasting this, it would probably make more sense to copy it somewhere shared first. But I would cross that bridge once another of our apps adopts this.

Reply:
> But I would suggest crossing that bridge once another of our apps adopts this.

Makes sense! Maybe @takameyer's React test apps could switch to making use of this for consistency?

Reply:
@tomduncalf Possibly, yes.  I actually am not using symbolic links.  I will have to test this out.